src/piece.c

```diff
@@ -262,23 +262,36 @@ pawn_valid(position pos, cell game[SIZE_STD][SIZE_STD], uint8_t * game_flags)
 	int		side_sign = 1;
 	if (game[pos.rank][pos.file].side == BLACK)
 		side_sign = -1;
-	if (game[pos.rank + side_sign][pos.file].piece == NULL)
+	if (game[pos.rank + side_sign][pos.file].piece == NULL 
+		&& pos.rank + side_sign >= 0 
+		&& pos.file >= 0 
+		&& pos.rank + side_sign < SIZE_STD 
+		&& pos.file < SIZE_STD){
 		valid_moves[move_idx++] = coords_to_pos(pos.rank + side_sign, pos.file);
-	if (game[pos.rank][pos.file].flags & FLAG_FIRSTMOVE && game[pos.rank + 2 * side_sign][pos.file].piece == NULL && game[pos.rank + 1 * side_sign][pos.file].piece == NULL) {
+		}
+	if (game[pos.rank][pos.file].flags & FLAG_FIRSTMOVE 
+		&& game[pos.rank + 2 * side_sign][pos.file].piece == NULL 
+		&& game[pos.rank + 1 * side_sign][pos.file].piece == NULL){
 		valid_moves[move_idx++] = coords_to_pos(pos.rank + 2 * side_sign, pos.file);
-	}
-	if (game[pos.rank + side_sign][pos.file + side_sign].piece != NULL &&
-	    game[pos.rank + side_sign][pos.file + side_sign].side ==
-	    opposite(game[pos.rank][pos.file].side)) {
+		}
+	if (game[pos.rank + side_sign][pos.file + side_sign].piece != NULL 
+		&& game[pos.rank + side_sign][pos.file + side_sign].side == opposite(game[pos.rank][pos.file].side)
+		&& pos.rank + side_sign >= 0 
```

Comment:
Not needed, we already check if the target cell has a piece, so checking for out of bounds in this case is redundant as (ideally) no piece should be out of bounds in the first place

Reply:
got it thanks